[Douglas Wei Jing Allwood] iP

douglaswja opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

douglaswja committed:
Completes Level-1

douglaswja committed:
Make the termination string a final variable

douglaswja committed:
Completes Level-2

douglaswja committed:
Include javadoc comments and some minor edits

douglaswja committed:
Completes Level-3

douglaswja committed:
Completes Level-4

douglaswja committed:
Level-4

douglaswja committed:
Updates the runtest.sh file to remove windows white spaces

douglaswja committed:
Include automated testing

douglaswja committed:
Corrects automated testing input file

douglaswja committed:
Level-5

douglaswja committed:
Level-6

douglaswja committed:
Updates the Command.java Enum to include all possible commands

The previous version separated commands and adding new tasks, this
version includes all possible DukeBot functionality as commands.

douglaswja committed:
Improves the code design and structure

douglaswja committed:
Include Javadoc comments for all relevant code

Additionally, this commit allows the TaskManager to be re-activated
after it has been deactivated.

douglaswja committed:
Updates javadoc comments

douglaswja committed:
Includes javadoc comment for TaskManager's listAll method

douglaswja committed:
Fixes automated CLI Text testing

douglaswja committed:
Includes comment on directory path finding for runtest.sh

douglaswja committed:
Change code style of Java Switch statements

douglaswja committed:
Include java package structure into Duke

douglaswja committed:
Implement download data but not save data

DukeBot CLI does not read or store data locally, persisting data for
only as long as it the program is running.

By not saving data, the DukeBot cannot be used across different
sessions.

Add a way to read data in from a text file.

This commit is incomplete in that it does not include a way to write
data to a text file.

The ability to write to a file requires us to store the Tasks completion
status, something that needs to be further planned out to utilise
existing code.

douglaswja committed:
Persist the task list in memory and load from memory on start-up

DukeBot runs with the Task List in RAM and does not persist it beyond
each session.

Temporary memory prevents the task list from being useful to a user
beyond a single session, requiring the user to keep DukeBot running.

Updates DukeBot to store the contents of its taskList to a file in
memory, allowing information to be persisted across multiple sessions.

There are some remaining To Do's involving refactoring the IO code into
separate functions and updating the Java Docs.

douglaswja committed:
Use LocalDate objects to store the /at and /by string

Task subclasses Event and Deadline treat the /at and /by, respectively,
as Strings only, preventing any datetime manipulation or behaviour.

The /at and /by inputs are likely to be dates and by treating them as
such future functionality such as reminders or listing all tasks for a
specific day or showing overdue tasks can be incorporated.

This commit updates the /at and /by instance variables to be stored as
LocalDate objects rather than String objects.

By using LocalDate objects we are able to perform many useful date
operations on the /at and /by instance variables.

douglaswja committed:
Merge branch 'branch-Level-7'

douglaswja committed:
Merge branch 'branch-Level-8'

douglaswja committed:
Refactor the code into OOP classes for UI, Parser, etc.

Each class in DukeBot handles its own IO and Parsing causing code to be
harder to maintain in the long run.

Spreading the IO and Parsing across many different classes makes DukeBot more
difficult to maintain in the long run.

Includes dedicated classes to handle these responsibilities.

All classes are now self-contained and hence any future changes to the
codebase can be targeted at whichever class is solely responsible for
the feature.

The Exception handling functionality, including the use of try-catch
blocks is not very elegant.

douglaswja committed:
Refactor Parser package to its parent directory

douglaswja committed:
Identify error in automated CLI test

douglaswja committed:
Import single classes instead of wildcard

douglaswja committed:
TaskList: Is-a relationship with ArrayList instead of Has-A

douglaswja committed:
Include JUnit tests for Storage and Parser

DBot does not have any JUnit unit tests for any classes or scenarios.

The lack of unit testing allows for potential regressions in any future
edits to the code base.

Includes JUnit tests for Storage initialisation on invalid file paths
and Parser.parse on a valid Todo command.

These tests are not comprehensive and more test cases are needed to
prevent regressions.

douglaswja committed:
JAR: .gitignore META-INF directory

STORAGE: Include directory existence and file existence checks that will
create the necessary directories and file if missing.

douglaswja committed:
Ui: Improve Help message and include Ui JavaDoc

douglaswja committed:
Include JavaDoc for all of DBot's Classes and Methods

DBot's implementation does not have many JavaDoc comments to explain
what each class or method is responsible for, how they are to be used,
or what the arguments they take represent.

This lack of JavaDoc comments can make it difficult for future
developers to use the DBot package.

Let's include JavaDoc comments for every Class or Method to assist
future developers.

JavaDoc comments can also be used by IDE's to quickly supply developers
with the information they need when dealing with a class or method.

douglaswja committed:
Command: Implement Find Command

douglaswja committed:
Merge branch 'branch-Level-9'

douglaswja committed:
FindCommand: Update quietExecute

douglaswja committed:
Ui: Update Help message

douglaswja committed:
FindCommand: Include JavaDoc comment for class initializer

CSjiade commented on src/main/java/dbot/command/Command.java:
```diff
@@ -0,0 +1,150 @@
+     *
+     * @param targetIndex A integer specifying the new target index for the 0-indexed TaskList.
+     */
+    protected void setTargetIndex(int targetIndex) {
```
Hi sir for this function should it be public too just like the rest of the functions?

PS: I may be wrong here

douglaswja replied:
Hi, thank you for the review.
You are perhaps right that the Setter can be public, ill look into changing it.

CSjiade commented on src/main/java/dbot/command/Command.java:
```diff
@@ -0,0 +1,150 @@
+     *
+     * @return An integer specifying the target index for the 0-indexed TaskList.
+     */
+    protected int getTargetIndex() {
```
Hi sir for this function should it be public too just like the rest of the functions?

PS: I may be wrong here

douglaswja replied:
Yes, ill look into changing this as well, thank you.

douglaswja committed:
Implement minor Java Coding Standard improvements

CSjiade commented on src/main/java/dbot/storage/Storage.java:
```diff
@@ -0,0 +1,148 @@
+     * @param tasks The TaskList to be saved.
+     * @throws StorageOperationException If the file cannot be written to.
+     */
+    public void save(TaskList tasks) throws StorageOperationException {
```
Will saveTasks be better here?

douglaswja replied:
Yes, that would be a more expressive name for the method and ill amend it accordingly.

CSjiade reviewed (commented):
Overall the code is very neat and good use of classes and oop. I think it's all good! Dont really have to edit anything. Good job! :)

spzj commented on src/main/java/dbot/command/Command.java:
```diff
@@ -0,0 +1,150 @@
+    /**
+     * Returns true if this Command is an Exit Command and false otherwise.
+     *
+     * @return True if this is an Exit Command and false otherwise.
```
Hi Douglas! You might want to edit the javadocs for this since it only returns false.

douglaswja replied:
Hi, thank you for the review.
This method was defined to return false in the Command class so that all classes that inherit it would also return false unless they override it, which is done only in the ExitCommand class.

I do understand the confusion that this method can cause and recognise that a more elegant solution might be to use a static method in the ExitCommand class to check if the user input is equivalent to an exit command.

(An example implementation that gave me the above idea can be found at: https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/commands/ExitCommand.java)

spzj commented on src/main/java/dbot/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Override
+    public void quietExecute(TaskList tasks, Storage storage) throws DBotException {
```
Since it's not a DBotException, you can remove "throws DBotException".

douglaswja replied:
Thank you for pointing this out! Might have missed this, will look into it.

spzj reviewed (commented):
Generally, I find your code really neat and well-documented. Great job!

douglaswja committed:
Merge remote-tracking branch 'origin/add-gradle-support'

douglaswja committed:
Includes gradle support

douglaswja committed:
Save for debugging purposes

douglaswja committed:
Checkstyle: Check capitalization of constant names for private attrs

douglaswja committed:
Correct the spelling of suppressions.xml (originally supressions.xml)

douglaswja committed:
Add GUI without modifying DBot

douglaswja committed:
Update DBot to support GUI - Cannot Open Display error

douglaswja committed:
Fix the Unable to open DISPLAY error

Intellij is able to build and run DBot.
`./gradlew run` still encounters the same issue but this is likely due
to permissions on Ubuntu.

See here for more info:
https://stackoverflow.com/questions/47509287/run-javafx-jar-as-sudo-unable-to-open-display

douglaswja committed:
MainWindow: Terminate on exit command and include welcome message

douglaswja committed:
List Tasks on separate lines upon `list` command

The `list` command displays all stored tasks on a single line which
makes it difficult to differentiate tasks from one another.

Lets display each task on a separate line and also include a list index
at the start of the line, e.g. "1. ".

douglaswja committed:
Address PR Review comments

douglaswja committed:
Address PR Review: Remove isExit from Command and its subclasses

douglaswja committed:
A-Assertions: Include assertion statements

douglaswja committed:
Parser: Refactor to improve code quality and incorporate Java streams

douglaswja committed:
Merge pull request #2 from douglaswja/branch-A-Assertions

A-Assertions: Include assertion statements

douglaswja committed:
Merge branch 'master' into branch-A-CodeQuality

douglaswja committed:
Merge pull request #3 from douglaswja/branch-A-CodeQuality

Branch a code quality

douglaswja committed:
A-UserGuide
